tools/wptrunner/wptrunner/testdriver-extra.js

```diff
@@ -125,4 +125,67 @@
         window.__wptrunner_message_queue.push({"type": "action", "action": "generate_test_report", "message": message});
         return pending_promise;
     };
+
+    window.test_driver_internal.add_virtual_authenticator = function(config) {
+        const pending_promise = new Promise(function(resolve, reject) {
+            pending_resolve = resolve;
+            pending_reject = reject;
+        });
+        window.__wptrunner_message_queue.push({"type": "action", "action": "add_virtual_authenticator", "config": config});
+        return pending_promise;
+    };
+
+    window.test_driver_internal.remove_virtual_authenticator = function(authenticator_id) {
+        const pending_promise = new Promise(function(resolve, reject) {
+            pending_resolve = resolve;
+            pending_reject = reject;
+        });
+        window.__wptrunner_message_queue.push({"type": "action", "action": "remove_virtual_authenticator", authenticator_id});
+        return pending_promise;
+    };
+
+    window.test_driver_internal.add_credential = function(authenticator_id, credential) {
+        const pending_promise = new Promise(function(resolve, reject) {
+            pending_resolve = resolve;
+            pending_reject = reject;
+        });
+        window.__wptrunner_message_queue.push({"type": "action", "action": "add_credential", authenticator_id, credential});
```

Comment:
I'd rather use the shorthand `{ something }` here since these lines are rather long. For consistency, I changed `config` on L134 to use the shorthand syntax as well.